src/Lean/Server/Completion.lean

```diff
@@ -26,14 +26,13 @@ builtin_initialize completionBlackListExt : TagDeclarationExtension ← mkTagDec
 def addToBlackList (env : Environment) (declName : Name) : Environment :=
   completionBlackListExt.tag env declName
 
-private def isBlackListed (declName : Name) : MetaM Bool := do
-  let env ← getEnv
-  (pure (declName.isInternal && !isPrivateName declName))
-  <||> (pure <| isAuxRecursor env declName)
-  <||> (pure <| isNoConfusion env declName)
-  <||> isRec declName
-  <||> (pure <| completionBlackListExt.isTagged env declName)
-  <||> isMatcher declName
+def isBlackListed (env : Environment) (declName : Name) : Bool :=
+  (declName.isInternal && !isPrivateName declName)
+  || isAuxRecursor env declName
+  || isNoConfusion env declName
+  || isRecCore env declName
+  || completionBlackListExt.isTagged env declName
+  || isMatcherCore env declName
```

Comment:
I was quite unsure about this change. I needed this to be outside of `MetaM` but I'm not certain about the extent to which Lean is now eagerly evaluating the components of this big chained or -- ideally it should stop after the first one that returns true (which I assume was certainly the case in the previous `MetaM` version).

Reply:
That's fine, `||` is lazy